src/topology/test/doesnt_reload.rs

```diff
@@ -0,0 +1,37 @@
+use crate::{
+    config::Config,
+    sinks::console::{ConsoleSinkConfig, Encoding, Target},
+    sources::socket::SocketConfig,
+    test_util::{next_addr, start_topology},
+};
+use std::path::Path;
+
+#[tokio::test]
+async fn topology_doesnt_reload_new_data_dir() {
+    let mut old_config = Config::builder();
+    old_config.add_source("in", SocketConfig::make_basic_tcp_config(next_addr()));
+    old_config.add_sink(
+        "out",
+        &["in"],
+        ConsoleSinkConfig {
+            target: Target::Stdout,
+            encoding: Encoding::Text.into(),
+        },
+    );
+    old_config.global.data_dir = Some(Path::new("/asdf").to_path_buf());
+    let mut new_config = old_config.clone();
+
+    let (mut topology, _crash) = start_topology(old_config.build().unwrap(), false).await;
+
+    new_config.global.data_dir = Some(Path::new("/qwerty").to_path_buf());
+
+    topology
+        .reload_config_and_respawn(new_config.build().unwrap())
+        .await
+        .unwrap();
+
+    assert_eq!(
+        topology.config.global.data_dir,
+        Some(Path::new("/asdf").to_path_buf())
+    );
```

Comment:
I have to admit, this doesn't make sense to me. Why wouldn't the updated configuration be present?  Kind of a weird test at face value.

Reply:
I don't know my own self. I do intend to take a pass over the tests in future work, so I've made a note of this question.

Reply:
We explicitly don't want to support changing the data directory at runtime. Original PR is [here](https://github.com/timberio/vector/pull/193) for (some) context.